### PR TITLE
Added LCM with test case

### PR DESCRIPTION
**Pull request by anandsit043**
Added definition for LCM in math.hpp and test cases. Having confusion in LCM(0, a) assuming it as 0. Please review.

### CODE REVIEW

**Comment by anandsit043** [comment]
Hi @faheel, Can you please check point where is this check failing?

File: include/functions/math.hpp [hunk 1]
```diff
@@ -182,4 +182,59 @@ BigInt gcd(const long long& num1, const BigInt& num2){
...
+    --------------------
+*/
+
+BigInt lcm(const BigInt &num1, const BigInt &num2){
+    BigInt abs_num1 = abs(num1);
+    BigInt abs_num2 = abs(num2);
+
+    // Trivial Case handling
+    if(abs_num1 == 0 && abs_num2 == 0)
+        throw std::logic_error("LCM undefined: Both number are zero");
```
Comment (faheel) [inline include/functions/math.hpp]: `lcm(0, 0)` exists, and is `0`.

File: include/functions/math.hpp [hunk 2]
```diff
@@ -182,4 +182,59 @@ BigInt gcd(const long long& num1, const BigInt& num2){
 BigInt gcd(const std::string& num1, const BigInt& num2){
     return gcd(num2, BigInt(num1));
 }
+
+/*    
+    lcm (BigInt, BigInt) :
+    --------------------
+*/
+
+BigInt lcm(const BigInt &num1, const BigInt &num2){
```
Comment (faheel) [inline include/functions/math.hpp]: The `lcm()` function can be written more simply as just this line:
```C++
    return abs(num1 * num2) / gcd(num1, num2);
```

Comment (anandsit043) [reply include/functions/math.hpp]: Done. Please review once.

Comment (faheel) [reply include/functions/math.hpp]: The single line above is the most efficient as there are no conditions being checked, and has the least number of temporary variables.

Comment (anandsit043) [reply include/functions/math.hpp]: @faheel 
Ok thanks.
Do I need to do something more in this. I have updated this as asked.

File: test/functions/math.test.cpp [hunk 3]
```diff
@@ -309,3 +309,62 @@ TEST_CASE("gcd()", "[functions][math][gcd][string]") {
...
 }
+
+/*
+    Naive lcm for testing purpose.
+*/
+
+long long naive_lcm(long long a, long long b){    
+    a = abs(a);
+    b = abs(b);
+
```
Comment (faheel) [inline test/functions/math.test.cpp]: This function also needs to be simplified.

File: include/functions/math.hpp [hunk 4]
```diff
@@ -182,4 +182,51 @@ BigInt gcd(const long long& num1, const BigInt& num2){
...
+    // Trivial Case handling
+    if(abs_num1 == 0 || abs_num2 == 0) return 0;
```
Comment (faheel) [inline include/functions/math.hpp]: This is not required.

File: include/functions/math.hpp [hunk 5]
```diff
@@ -182,4 +182,51 @@ BigInt gcd(const long long& num1, const BigInt& num2){
...
+
+    return abs(num1 * num2)/gcd(abs_num1, abs_num2);
```
Comment (faheel) [inline include/functions/math.hpp]: `abs_` prefix needs to be removed.

File: include/functions/math.hpp (hunk 4 above)
Comment (anandsit043) [reply include/functions/math.hpp]: if both are zero then gcd is zero and it removing this gives  division by zero error.

Comment (faheel) [reply include/functions/math.hpp]: Ah, yes, I missed this point. But don't forget to remove the `abs_` prefix or it won't compile.

**Comment by anandsit043** [comment]
Done.

**Comment by anandsit043** [comment]
Hi @faheel ,
Didn't heard from you. So replying again. Please look at the modified changes.

Thanks

**Comment by faheel** [comment]
Sorry for the late review, I was a little busy. Thank you for contributing! 🙂

**Pull request merged**